src/azure-cli/azure/cli/command_modules/netappfiles/tests/latest/test_volume_commands.py

```diff
@@ -84,21 +96,75 @@ def test_create_volume_with_subnet_in_different_rg(self):
+        rg_r = self.create_random_name(prefix='cli-rg-', length=24)
+        subs_id = self.current_subscription()
+        self.cmd("az group create -n %s --subscription %s -l %s" % (rg_r, subs_id, DP_RG_LOCATION)).get_output_in_json()
+
+        self.setup_vnet(rg_r, vnet_name, subnet_name, '10.1.0.0', DP_RG_LOCATION)
+        self.cmd("az netappfiles account create -g %s -a %s -l %s" % (rg_r, account_name_r, DP_ANF_LOCATION)).get_output_in_json()
+        self.cmd("az netappfiles pool create -g %s -a %s -p %s -l %s %s" % (rg_r, account_name_r, pool_name_r, DP_ANF_LOCATION, POOL_DEFAULT)).get_output_in_json()
+
+        subnet_id = "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/%s/subnets/%s" % (subs_id, rg_r, vnet_name, subnet_name)
+
+        # volume = self.cmd("az netappfiles volume create --resource-group %s --account-name %s --pool-name %s --volume-name %s -l %s %s --file-path %s --vnet %s --subnet %s" % (rg_r, account_name_r, pool_name_r, volume_name_r, DP_ANF_LOCATION, VOLUME_DEFAULT, file_path, vnet_name, subnet_id)).get_output_in_json()
+
+        dst_volume = self.cmd("az netappfiles volume create --resource-group %s --account-name %s --pool-name %s --volume-name %s -l %s %s --file-path %s --vnet %s --subnet %s --volume-type %s --endpoint-type %s --replication-schedule %s --remote-volume-resource-id %s" % (rg_r, account_name_r, pool_name_r, volume_name_r, DP_ANF_LOCATION, VOLUME_DEFAULT, file_path, vnet_name, subnet_id, "DataProtection", "dst", "_10minutely", src_volume['id'])).get_output_in_json()
+        assert dst_volume['dataProtection'] is not None
+        assert dst_volume['id'] is not None
+        time.sleep(2)
+
+        # approve
+        self.cmd("az netappfiles volume replication approve -g %s -a %s -p %s -v %s --remote-volume-resource-id %s" % ('{rg}', account_name, pool_name, volume_name, dst_volume['id']))
+        self.wait_for_replication_status("Mirrored", rg_r, account_name_r, pool_name_r, volume_name_r)
+
+        # break
+        self.cmd("az netappfiles volume replication suspend -g %s -a %s -p %s -v %s" % (rg_r, account_name_r, pool_name_r, volume_name_r))
+        self.wait_for_replication_status("Broken", rg_r, account_name_r, pool_name_r, volume_name_r)
+
+        # resume
+        self.cmd("az netappfiles volume replication resume -g %s -a %s -p %s -v %s" % (rg_r, account_name_r, pool_name_r, volume_name_r))
+        self.wait_for_replication_status("Mirrored", rg_r, account_name_r, pool_name_r, volume_name_r)
+
+        # break
+        self.cmd("az netappfiles volume replication suspend -g %s -a %s -p %s -v %s" % (rg_r, account_name_r, pool_name_r, volume_name_r))
+        self.wait_for_replication_status("Broken", rg_r, account_name_r, pool_name_r, volume_name_r)
+
+        # delete
+        self.cmd("az netappfiles volume replication remove -g %s -a %s -p %s -v %s" % (rg_r, account_name_r, pool_name_r, volume_name_r))
+        time.sleep(2)
```

Comment:
why sleep here?

Reply:
I think this is needed to ensure this completes before any cleanup after else it mayl try and delete the resources before they are ready.